src/diffusers/pipelines/pipeline_loading_utils.py

```diff
@@ -157,13 +156,21 @@ def variant_compatible_siblings(filenames, variant=None) -> Union[List[os.PathLi
     transformers_index_format = r"\d{5}-of-\d{5}"
 
     if variant is not None:
-        # `diffusion_pytorch_model.fp16.bin` as well as `model.fp16-00001-of-00002.safetensors`
+        # Examples:
+        # For transformers, `pytorch_model.fp16.bin` as well as `pytorch_model.fp16-00001-of-00004.bin`.
+        # For diffusers `diffusion_pytorch_model.fp16.bin` as well as `diffusion_pytorch_model-00001-of-00002.fp16.safetensors`
+        # These differences exist because `diffusers` delegates the process of loading sharded checkpoints
+        # to `accelerate`. However, `transformers` has custom code that takes care of it.
```

Comment:
More specifically, we rely on 

https://github.com/huggingface/diffusers/blob/3f1411767bc0f1837adb6f289713807f18599db3/src/diffusers/models/modeling_utils.py#L774

which expects the index file such that its name always ends with "index.json". 

`transformers` handles the loading of sharded checkpoints differently. This is why we need to have code to handle the nuances here.